box-share-sdk/src/main/java/com/box/androidsdk/share/fragments/InviteCollaboratorsFragment.java

```diff
@@ -0,0 +1,276 @@
+import com.box.androidsdk.content.models.BoxFolder;
+import com.box.androidsdk.content.models.BoxSession;
+import com.box.androidsdk.content.models.BoxUser;
+import com.box.androidsdk.content.requests.BoxRequestBatch;
+import com.box.androidsdk.content.requests.BoxRequestsShare;
+import com.box.androidsdk.content.requests.BoxResponse;
+import com.box.androidsdk.content.requests.BoxResponseBatch;
+import com.box.androidsdk.content.utils.BoxLogUtils;
+import com.box.androidsdk.content.utils.SdkUtils;
+import com.box.androidsdk.internal.BoxApiInvitee;
+import com.box.androidsdk.internal.BoxInviteeResponse;
+import com.box.androidsdk.share.CollaborationUtils;
+import com.box.androidsdk.share.R;
+import com.box.androidsdk.share.adapters.InviteeAdapter;
+import com.box.androidsdk.share.internal.BoxListInvitees;
+
+import java.net.HttpURLConnection;
+import java.util.Locale;
+
+public class InviteCollaboratorsFragment extends BoxFragment implements View.OnClickListener, CollaborationRolesDialog.OnRoleSelectedListener {
+
+    protected static final String TAG = InviteCollaboratorsFragment.class.getName();
+    public static final String EXTRA_ACCESS_TOKEN = "InviteCollaboratorsFragment.ExtraAccessToken";
+    private Button mRoleButton;
+    private MultiAutoCompleteTextView mAutoComplete;
+    private InviteeAdapter mAdapter;
+    private BoxCollaboration.Role mSelectedRole;
+    private BoxCollaboration.Role[] mRoles;
+    private String mAccessToken;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_invite_collaborators, container, false);
+
+        mRoleButton = (Button) view.findViewById(R.id.invite_collaborator_role);
+        mRoleButton.setOnClickListener(this);
+        mAutoComplete = (MultiAutoCompleteTextView) view.findViewById(R.id.invite_collaborator_autocomplete);
+        mAutoComplete.setTokenizer(new MultiAutoCompleteTextView.CommaTokenizer());
+        mAdapter = new InviteeAdapter(getActivity());
+        mAutoComplete.setAdapter(mAdapter);
+
+        if (getArguments() != null) {
+            Bundle args = getArguments();
+            mAccessToken = args.getString(EXTRA_ACCESS_TOKEN);
+        }
+
+        // Get serialized roles or fetch them if they are not available
+        if (getFolder() != null && getFolder().getAllowedInviteeRoles() != null) {
+            mRoles = getFolder().getAllowedInviteeRoles().toArray(new BoxCollaboration.Role[getFolder().getAllowedInviteeRoles().size()]);
+            BoxCollaboration.Role selectedRole = mRoles[0];
+            setSelectedRole(selectedRole);
+        } else {
+            fetchRoles();
+        }
+
+        fetchInvitees();
+        return view;
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        // Inflate the menu; this adds items to the action bar if it is present.
```

Comment:
If we are handling menu items in the activity we should be consistent. 
I don't think it is a good idea to mix and match between fragments and activities as it will add confusion.